internal/providers/terraform/aws/neptune_cluster_instance.go

```diff
@@ -0,0 +1,86 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetNeptuneClusterInstanceRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_neptune_cluster_instance",
+		RFunc: NewNeptuneClusterInstance,
+	}
+}
+
+func NewNeptuneClusterInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	fmt.Println(d)
+	var monthlyCPUCreditHrs *decimal.Decimal
+	region := d.Get("region").String()
+	instanceClass := d.Get("instance_class").String()
+	hourlyQuantity := 1
+	if d.Get("count").Type != gjson.Null {
+		hourlyQuantity = int(d.Get("count").Int())
+	}
+
+	if u != nil && u.Get("monthly_cpu_credit_hrs").Type != gjson.Null {
+		monthlyCPUCreditHrs = decimalPtr(decimal.NewFromInt(u.Get("monthly_cpu_credit_hrs").Int()))
+	}
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	if instanceClass == "db.t3.medium" {
+		costComponents = append(costComponents, neptuneClusterCPUInstanceCostComponent(region, instanceClass, monthlyCPUCreditHrs))
+	} else {
+		costComponents = append(costComponents, neptuneClusterDbInstanceCostComponent(instanceClass, region, instanceClass, hourlyQuantity))
+	}
```

Comment:
This logic is slightly wrong, for t3, we want to show the instance hourly cost AND the cpu credits (using the logic `(strings.HasPrefix(instanceType, "db.t3.")`)

`db.products.distinct("attributes.usagetype", {"vendorName": "aws", "service":"AmazonNeptune"});` shows that there are two usagetypes with the word "cpu" or "credit":
1. SAE1-CPUCredits:db.t3
2. APE1-CPUCredits:db.t3

This is odd, but then AWS' pricing is sometimes odd :) So I think we can use the usage type regex `CPUCredits:db.t3` to find the price, as they both have the same price (USD 0.15 per vCPU-Hour).

Reply:
@alikhajeh1 added changes , check plz